File: Programmers/Level 1/2.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    string answer = "";
    int sz = s.size();
    if (sz % 2 == 0) {
       //짝수
        answer += s.at(sz/2-1);
        answer += s.at(sz/2);
    }
    else {
        //홀수
        answer = s.at(sz/2);
    }
    return answer;
}

File: Programmers/Level-4/2.py
def find_max(money):
    dp = [0] * 1000001
    dp[0] = money[0]
    dp[1] = max(money[0], money[1])
    for i in range(2, len(money) - 1): # 첫 집 무조건 터는 경우
        dp[i] = max(money[i]+dp[i-2], dp[i-1])
    return max(dp)

def solution(money):
    answer = find_max(money)
        
    money.reverse()
    
    answer = max(answer, find_max(money))
        
    return answer
